core/src/main/java/org/apache/carbondata/core/datamap/DataMapFilter.java

```diff
@@ -115,10 +139,12 @@ public DataMapFilter(FilterResolverIntf resolver) {
     this.resolver = resolver;
   }
 
-  private void resolve() {
+  public void resolve(boolean lazyResolve) {
     if (expression != null) {
-      table.processFilterExpression(expression, null, null);
-      resolver = CarbonTable.resolveFilter(expression, table.getAbsoluteTableIdentifier());
+      processFilterExpression();
+      if (!lazyResolve) {
```

Comment:
please add comment when lazyResolve will be true and false

Reply:
added in resolve method